src/app/static/src/app/store/modelRun/mutations.ts

```diff
@@ -1,46 +1,55 @@
-import {Mutation, MutationTree} from "vuex";
+import {MutationTree} from "vuex";
 import {ModelRunState} from "./modelRun";
 import {PayloadWithType} from "../../types";
 import {ModelResultResponse, ModelStatusResponse, ModelSubmitResponse} from "../../generated";
-import {readyStateMutations} from "../shared/readyStateMutations";
 
-type ModelRunMutation = Mutation<ModelRunState>
-
-export interface ModelRunMutations {
-    ModelRunStarted: ModelRunMutation
-    RunStatusUpdated: ModelRunMutation,
-    PollingForStatusStarted: ModelRunMutation,
-    RunResultFetched: ModelRunMutation,
-    RunResultError: ModelRunMutation,
-    Ready: ModelRunMutation
+export enum ModelRunMutation {
+    ModelRunStarted="ModelRunStarted",
+    ModelRunError="ModelRunError",
+    RunStatusUpdated = "RunStatusUpdated",
+    PollingForStatusStarted = "PollingForStatusStarted",
+    RunResultFetched = "RunResultFetched",
+    RunResultError= "RunResultError",
+    RunStatusError="RunStatusError",
+    Ready= "Ready"
 }
 
-export const mutations: MutationTree<ModelRunState> & ModelRunMutations = {
-    ModelRunStarted(state: ModelRunState, action: PayloadWithType<ModelSubmitResponse>) {
+export const mutations: MutationTree<ModelRunState>  = {
+    [ModelRunMutation.ModelRunStarted](state: ModelRunState, action: PayloadWithType<ModelSubmitResponse>) {
         state.modelRunId = action.payload.id;
         state.status = {id: action.payload.id} as ModelStatusResponse;
         state.errors = [];
     },
 
-    RunStatusUpdated(state: ModelRunState, action: PayloadWithType<ModelStatusResponse>) {
+    [ModelRunMutation.RunStatusUpdated](state: ModelRunState, action: PayloadWithType<ModelStatusResponse>) {
         if (action.payload.done) {
             clearInterval(state.statusPollId);
             state.statusPollId = -1;
         }
         state.status = action.payload;
     },
 
-    PollingForStatusStarted(state: ModelRunState, action: PayloadWithType<number>) {
+    [ModelRunMutation.PollingForStatusStarted](state: ModelRunState, action: PayloadWithType<number>) {
         state.statusPollId = action.payload;
     },
 
-    RunResultFetched(state: ModelRunState, action: PayloadWithType<ModelResultResponse>) {
+    [ModelRunMutation.RunResultFetched](state: ModelRunState, action: PayloadWithType<ModelResultResponse>) {
         state.result = action.payload;
     },
 
-    RunResultError(state: ModelRunState, action: PayloadWithType<string>) {
+    [ModelRunMutation.RunResultError](state: ModelRunState, action: PayloadWithType<string>) {
         state.errors.push(action.payload);
     },
 
-    ...readyStateMutations
+    [ModelRunMutation.Ready](state: ModelRunState) {
+        state.ready = true;
+    },
+
+    [ModelRunMutation.ModelRunError](state: ModelRunState, action: PayloadWithType<string>) {
```

Comment:
I noticed that we were telling the api to commit these mutations on error, but they didn't actually exist! So I've migrated this mutation tree over to using the enum pattern, and I added a helper test method that checks that all the enum values correspond to methods: https://github.com/mrc-ide/hint/pull/173/files#diff-4f54fb77cd41d6dd4f4bed5bdac03397